dashboard/15-final/app/login/page.tsx

```diff
@@ -1,6 +1,13 @@
 import LoginForm from '@/app/ui/login-form';
+import { getServerSession } from 'next-auth/next';
+import { redirect } from 'next/navigation';
+import { authOptions } from '@/app/api/auth/[...nextauth]/route';
+
+export default async function Page() {
+  // Get user session token
+  const session = await getServerSession(authOptions);
```

Comment:
Could put this into to a root-level `auth.ts`, together with `authOptions`. See:

https://github.com/nextauthjs/next-auth/blob/5a7c1bb2bbdbba776f1b3e9228e6f3b9a287aea6/apps/examples/nextjs/auth.ts#L149-L153